### PR TITLE
prepare v3.5.0

### PR DESCRIPTION
**Pull request by idinium96**
## Added
- Options.json related:
    - ✨option to disable sending the pre-accepted message - @idinium96
        - new option added: `offerReceived.sendPreAcceptMessage.enable` (default is `true`)
    - ✨option to only show price update that is in stock - @idinium96
       - new option added: `discordWebhook.priceUpdate.showOnlyInStock` (default is `false`)
    - ✨an option to partially update item prices if in stock and the difference between the new selling price and current buying price is negative (#284 - read: https://github.com/TF2Autobot/tf2autobot/pull/337) - @idinium96
    - ✨ option to select highValue spells - @idinium96
        - new option added: `highValue.spells` (default is `[]` - all spells)
- Others:
     - ✨Docker build for node images (refer: [Wiki](https://github.com/TF2Autobot/tf2autobot/wiki/Running-the-bot-using-docker)) - @rennokki
     - ✨support **buy order** for skins/war paints (Festivized skins/war paints not supported - see [`#337`](https://github.com/TF2Autobot/tf2autobot/pull/377)) - @idinium96
    - ✨option to disable socket entirely (read: https://github.com/TF2Autobot/tf2autobot/pull/383) - @idinium96

## Changed/updated
- 🔄 replaced bracket with a dash to not combine with generated URL on retrying to accept offer message sent to admins via Steam Chat - @idinium96
- 🤝 custom-pricer friendly (#363) - @idinium96
- 📝 log bot uptime on auto-restart - @idinium96
- 🔑 ensure selling price for Mann Co. Supply Crate Key never be zero - @idinium96
- 🔄 updated dependencies - @idinium96
- 🔄 improve `!deposit` command (#325, #326) - @idinium96
- 🔄 more flexibility when adding items with parameters - @idinium96
- 🔄 refactor `!find` command (view what changed: https://github.com/TF2Autobot/tf2autobot/pull/387) - @idinium96
- ✅ do not hold items in an offer for review - @idinium96
- ⏮ revert only pick 1 craft weapon when constructing an offer - @idinium96

## Fixed
- 🔨 wrong position for pure stock and total items in the Offer review send as Discord Webhook.
- `!autoadd` command:
    - 🐛 fix the glitch where the pure got deleted from items dictionary - @idinium96
    - ⌚ add a delay before sending `autoadd` summary - @idinium96
- 🔨 fix generic unusual buy order (ensure the generic match is used against 'their' inventory) (#340) - @joekiller
- 🔨 fix sending duplicated trade summary - @idinium96
- 🔨 fix failed to disable Autokeys error - @idinium96
- 🐛 fix bot crashed (rare) on startup - @idinium96 
    - ⛔ do not execute sending price update to Discord Webhook if globalKeyPrices still undefined

### CODE REVIEW

**Comment by idinium96** [comment]
@rennokki Looks like it's working fine. 👍

**Comment by rennokki** [comment]
@idinium96 Shipping a product was never this easy. 👀

**Comment by idinium96** [comment]
> @idinium96 Shipping a product was never this easy. 👀

Haha. I will not merge this today. Probably in another 2 - 3 days.

**Comment by rennokki** [comment]
@idinium96 Not a problem, iterations vary. I'll make sure wiki is done until then.

**Comment by idinium96** [comment]
> @idinium96 Not a problem, iterations vary. I'll make sure wiki is done until then.

Thanks a bunch. I suggest you share with others first in the #contributors channel in our Discord server before publishing it.

**Review by arik123 — COMMENTED** [review]
this is just too much to review

**Comment by idinium96** [comment]
> this is just too much to review

I am sure everything here already reviewed, so probably best to just merge and ship v3.5.0?

**Review by arik123 — APPROVED** [review]
So I looked at highValue spells, and everything else should be reviewed so I approve

File: src/app.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ import { XMLHttpRequest } from 'xmlhttprequest-ts';
 import { uptime } from './lib/tools/time';
 
 ON_DEATH({ uncaughtException: true })((signalOrErr, origin) => {
-    const crashed = signalOrErr !== 'SIGINT';
+    const crashed = ['SIGINT', 'SIGTERM'].indexOf(signalOrErr as 'SIGINT' | 'SIGTERM' | 'SIGQUIT') === -1;
```
Comment (arik123) [inline src/app.ts:62]: you probably should use `includes` instead of `indexOf`

**Review by arik123 — CHANGES_REQUESTED** [review]

File: src/app.ts (hunk 1 above)
Comment (idinium96) [reply src/app.ts:62]: Yeah, I have that thought too. Will change it. Thanks.

**Review by arik123 — APPROVED** [review]

**Pull request merged**